wrappers/desktop.go

```diff
@@ -163,6 +164,15 @@ func sanitizeDesktopFile(s *snap.Info, rawcontent []byte) []byte {
 	return []byte(strings.Join(newContent, "\n"))
 }
 
+func updateDesktopDatabase() error {
+	if _, err := exec.LookPath("update-desktop-database"); err == nil {
```

Comment:
I'd add a `// TODO: switch on release.ReleaseInfo.ID to decide what distro-specific hook to run`